pymodbus/client/base.py

```diff
@@ -143,15 +143,17 @@ def execute(self, request: ModbusRequest | None = None):
         :returns: The result of the request execution
         :raises ConnectionException: Check exception text.
         """
-        if not self.transport:
+        if not self.transport and self.reconnect_task is None:
```

Comment:
> This looks wrong because there are only a queue for serial !

I think this should work for transports without queues as well although it seems to be more problematic with a queue, if we are reconnecting any requests need to wait for the re-connection to complete before they can execute.

> Apart from that, it is important that the APP is informed that we are in a non-operational state, remember reconnect times runs sometimes in hours.

Isn't it informed already via the `ModbusIOException` that gets thrown?

Reply:
> I think this should work for transports without queues as well although it seems to be more problematic with a queue, if we are reconnecting any requests need to wait for the re-connection to complete before they can execute.

It doesn´t at least not consistently.


> Isn't it informed already via the `ModbusIOException` that gets thrown?

That is exactly the point, the Exception do not get thrown when reconnect is in progress !

Reply:
> It doesn´t at least not consistently.

Hmm, what issues does it cause exactly?

> That is exactly the point, the Exception do not get thrown when reconnect is in progress !

Yeah, I mean it would be the read that triggered the reconnect essentially that you would use.

Reply:
This is still wrong because a reconnect can very well happen within the timeout period.

And with queues it is even more wrong, as one of the ideas of a queue is to buffer calls.